internal/latestrelease/store.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package latestrelease
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/file"
+	"github.com/mongodb/mongocli/internal/version"
+	"github.com/spf13/afero"
+)
+
+//go:generate mockgen -destination=../mocks/mock_version_store.go -package=mocks github.com/mongodb/mongocli/internal/latestrelease Store
+
+type State struct {
+	CheckedForUpdateAt   time.Time `yaml:"checked_for_update_at"`
+	LatestReleaseVersion string    `yaml:"latest_release"`
+}
+
+type BrewPath struct {
+	CheckedPathAt  time.Time `yaml:"checked_path_at"`
+	ExecutablePath string    `yaml:"path"`
+	HomePath       string    `yaml:"home_path"`
+}
+
+type Store interface {
+	LoadLatestVersion(tool string) (string, error)
+	SaveLatestVersion(tool, ver string) error
+	LoadBrewPath(tool string) (string, string, error)
+	SaveBrewPath(tool, execPath, homePath string) error
+}
+
+func NewStore(fileSystem afero.Fs) Store {
+	return &store{fs: fileSystem}
+}
+
+type store struct {
+	fs afero.Fs
+}
+
+const (
+	stateFileSubPath = "/rstate.yaml"
```

Comment:
this will be `.config/rstate.yaml` can you be certain no other app will use it?

Reply:
great point, for atlascli is ensured it will be within the atlascli folder, but it looks like for mcli we put into `.config`. Should I add a mongocli prefix to the file or do we plan to move the config to a mongocli folder? (I can do that change if we think we should)

Reply:
as we synced offline - https://jira.mongodb.org/browse/CLOUDP-114248, for now I'll add the /mongocli/ to the new file paths